worlds/yoshisisland/__init__.py

```diff
@@ -0,0 +1,410 @@
+import os
+import typing
+import threading
+import dataclasses
+
+from typing import Dict, List, Set, TextIO
+from BaseClasses import Item, MultiWorld, Location, Tutorial, ItemClassification
+from worlds.AutoWorld import World, WebWorld
+from Options import PerGameCommonOptions
+import Patch
+import settings
+from .Items import get_item_names_per_category, item_table, filler_items, trap_items
+from .Locations import get_locations
+from .Regions import create_regions
+from .Options import YoshisIslandOptions
+from .setup_game import setup_gamevars
+from .Client import YISNIClient
+from .Rules import set_easy_rules, set_normal_rules, set_hard_rules
+from .Rom import LocalRom, patch_rom, get_base_rom_path, YIDeltaPatch, USHASH
+
+class YISettings(settings.Group):
+    class RomFile(settings.SNESRomPath):
+        """File name of the Yoshi's Island 1.0 US rom"""
+        description = "Yoshi's Island ROM File"
+        copy_to = "Super Mario World 2 - Yoshi's Island (U).sfc"
+        md5s = [USHASH]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+
+class YIWeb(WebWorld):
+    theme = "ocean"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Yoshi's Island randomizer"
+        "and connecting to an Archipelago server.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Pink Switch"]
+    )
+
+    tutorials = [setup_en]
+
+class YIWorld(World):
+    """Yoshi's Island is a 2D platforming game.
+        During a delivery, Bowser's evil ward, Kamek, attacked the stork, kidnapping Luigi and dropping Mario onto Yoshi's Island.
+        As Yoshi, you must run, jump, and throw eggs to escort the baby Mario across the island to defeat Bowser and reunite the two brothers with their parents."""
+    game: str = "Yoshi's Island"
```

Comment:
```suggestion
    game = "Yoshi's Island"
```
Don't need to type it here